Framework/Algorithms/src/AddAbsorptionWeightedPathLengths.cpp

```diff
@@ -94,57 +98,79 @@ std::map<std::string, std::string> AddAbsorptionWeightedPathLengths::validateInp
  */
 void AddAbsorptionWeightedPathLengths::exec() {
 
-  const PeaksWorkspace_sptr inputWS = getProperty("InputWorkspace");
+  const API::IPeaksWorkspace_sptr inputWS = getProperty("InputWorkspace");
   const int nevents = getProperty("EventsPerPoint");
   const int maxScatterPtAttempts = getProperty("MaxScatterPtAttempts");
+  const int seed = getProperty("SeedValue");
+
+  PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(inputWS);
+  LeanElasticPeaksWorkspace_sptr lpws = std::dynamic_pointer_cast<LeanElasticPeaksWorkspace>(inputWS);
+
+  if (lpws && !pws) {
```

Comment:
Is it worth checking if the lpws has a valid instrument on it before creating this fake one?

So for example

```python
peaks = LoadIsawPeaks(Filename=r'Peaks5637.integrate')
leanpeaks = ConvertPeaksWorkspace(peaks)
```